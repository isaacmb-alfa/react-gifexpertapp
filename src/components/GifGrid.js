import React/*, { useState, useEffect }*/ from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from './../hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {

    //custom Hook
    const { data:images, loading} = useFetchGifs(category);

    // console.log(loading);



    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__slow animate__repeat-2 animate__headShake">Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
GifGrid.propTypes= {
    category: PropTypes.string.isRequired
}
