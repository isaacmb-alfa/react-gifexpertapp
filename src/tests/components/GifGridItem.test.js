import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';

describe('Prueba que <GifGridItem/> corra correctamente', () => {
        const title = 'Un titulo';
        const url = 'https://localhost/algo.jpg';
        const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('debe de mostrar <GifGridItem/> correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('debe de tener un parrafo con el title', () => {
        
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        // console.log(img.props());

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('debe de tener  animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        // console.log(div.html().includes('animate__fadeIn'));
        console.log(div.prop('className'));

        // expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
        expect(className.includes('animate__fadeIn')).toBe(true);

    });

});