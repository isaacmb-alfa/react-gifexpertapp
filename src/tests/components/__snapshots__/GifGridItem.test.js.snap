// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba que <GifGridItem/> corra correctamente debe de mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
